common.go

```diff
@@ -22,14 +23,14 @@ import (
 )
 
 const (
-	VersionSSL30        = 0x0300
```

Comment:
One of the main points is to keep this code as close as master branch in go lang, so that it's easy to upstream (that's we keep TLS 1.3 in separated file).
I understand where you want to go, but big refactoring is probably not what we need at this point.